### PR TITLE
feat(argo-workflows): Add support for extra containers in server & controller deployment

### PR DESCRIPTION
**Pull request by mikebryant**
My use-case for this is using persistence with a GCP CloudSQL database, using the sidecar pattern for the cloud-sql-proxy

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by mikebryant** [comment]
Some basic testing:
```
mikebryant@BULB-MB-0246 argo-workflows % cat env.yaml
server:
  extraContainers:
    - name: blah
      image: blah
mikebryant@BULB-MB-0246 argo-workflows % helm template . --values env.yaml | tail
          - name: tmp
            mountPath: /tmp

        - image: blah
          name: blah
      volumes:
      - name: tmp
        emptyDir: {}
      nodeSelector:
        kubernetes.io/os: linux
```

File: charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml [hunk 1]
```diff
@@ -79,6 +79,9 @@ spec:
               containerPort: {{ .Values.controller.metricsConfig.port }}
             - containerPort: 6060
           livenessProbe: {{ .Values.controller.livenessProbe | toYaml | nindent 12 }}
+        {{- if .Values.controller.extraContainers }}
+        {{ toYaml .Values.controller.extraContainers | nindent 8}}
+        {{- end }}
```
Comment (oliverbaehler) [inline charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml:84]:
```suggestion
        {{- with .Values.controller.extraContainers }}
          {{- toYaml . | nindent 8 }}
        {{- end }}
```

File: charts/argo-workflows/templates/server/server-deployment.yaml [hunk 2]
```diff
@@ -80,6 +80,9 @@ spec:
           {{- with .Values.server.volumeMounts }}
             {{- toYaml . | nindent 10}}
           {{- end }}
+        {{- if .Values.server.extraContainers }}
+        {{ toYaml .Values.server.extraContainers | nindent 8}}
+        {{- end }}
```
Comment (oliverbaehler) [inline charts/argo-workflows/templates/server/server-deployment.yaml:85]:
```suggestion
        {{- with .Values.server.extraContainers }}
          {{- toYaml . | nindent 8 }}
        {{- end }}
```

**Review by oliverbaehler — CHANGES_REQUESTED** [review]

**Comment by mikebryant** [comment]
Thanks for the review, have applied the requested changes :)

**Review by oliverbaehler — APPROVED** [review]
lgtm

**Pull request merged**